shell/common/gin_helper/wrappable.cc

```diff
@@ -69,8 +70,16 @@ void WrappableBase::FirstWeakCallback(
 // static
 void WrappableBase::SecondWeakCallback(
     const v8::WeakCallbackInfo<WrappableBase>& data) {
-  WrappableBase* wrappable = data.GetParameter();
-  delete wrappable;
+  // Certain classes (for example api::WebContents and api::WebContentsView)
+  // are running JS code in the destructor, while V8 may crash when JS code
+  // runs inside weak callback.
+  //
+  // We work around this problem by delaying the deletion to next tick where
+  // garbage collection is done.
+  base::ThreadTaskRunnerHandle::Get()->PostNonNestableTask(
+      FROM_HERE,
+      base::BindOnce([](WrappableBase* wrappable) { delete wrappable; },
```

Comment:
Do we need to check the return value of `PostNonNestableTask` and incase its `false` we have to finish things synchronously. But that could only happen during shutdown in which case leaking objects are fine.

Reply:
Objects that must be deleted on exit are tracked by `TrackableObject`, other objects are free to leak on exit. V8 does not collect all native objects on exit so leaking is a quite common thing.